### ISSUE TITLE
Variable in condition outside a lock is modified later in a lock

### ISSUE THREAD

**Issue by tvsonar** [SonarSource-VisualStudio/sonarlint-roslyn-analyzer#346]
```
if (a == 0)
  return;

lock(obj)
{
  a--;
}
```

Do not report on double checked locking.
